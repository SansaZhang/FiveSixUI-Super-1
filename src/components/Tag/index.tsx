import React, { PropTypes } from 'react';
import classNames from 'classnames';


export interface TagProps {
  prefixCls?: string;
  value?: any;
  /** 标签是否能关闭 */
  closable?: boolean;
  /** 标签是否选中 */
  selected?: boolean;
  /** 点击时的回调 */
  onClick?:Function;
  /** 关闭时的回调 */
  onClose?: Function;
}

export default class Tag extends React.Component<TagProps, any> {
  static defaultProps = {
    prefixCls: 'wl-tag',
    selected: false,
    closable: true,
  };

  static propTypes = {
    prefixCls: PropTypes.string,
    value: PropTypes.any.isRequired,
    onClick: PropTypes.func,
    onClose: PropTypes.func,
    selected: PropTypes.bool,
    closable: PropTypes.bool,
  };

  handlerClick = (e,value) => {
    e && e.stopPropagation();
    const { onClick, selected } = this.props;
    onClick && onClick(value, !selected);
  }

  handlerClose = (e,value) => {
    e && e.stopPropagation();
    const { onClose } = this.props;
    onClose && onClose(value);
  }

  render() {
    const {prefixCls, value, selected, closable, className, children,...other} = this.props;
    const closeIcon = closable ? <span className={`${prefixCls}-close`} onClick={e => this.handlerClose(e, value)} /> : '';
    const classString = classNames({
      [prefixCls]: true,
      [`${prefixCls}-selected`]: selected,
      [className]: !!className
    });
    return (
       <div 
        data-value={value}
        className={classString}
        onClick={e => this.handlerClick(e, value)}
        {...other}
      >
        <span className={`${prefixCls}-text`}>{children}</span>
        {closeIcon}
      </div>
    );
  }
}
