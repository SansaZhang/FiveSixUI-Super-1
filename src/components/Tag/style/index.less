@import '../../style/index.less';
@import "../../style/themes/default";
@import "../../style/mixins/index";

@tag-prefix-cls: ~"@{wl-prefix}-tag";

.@{tag-prefix-cls} {
    display: inline-block;
    background: #f2f2f2;
    border: 1px solid #cfcfcf;
    border-radius: 3px;
    font-size: @font-size-base;
    line-height: 20px;
    padding: 0 3px;
    cursor: default;

    &-text {
        color: #666;
    }

    &-close {
        display:inline-block; 
        width:12px; 
        height:2px; 
        background:#f00; 
        font-size:0; 
        line-height:0;
        vertical-align:middle;
        margin-top: -2px;
        margin-left: 5px;
        cursor: pointer;
        -webkit-transform: rotate(45deg);
    }

    &-close:after { 
        content:'.'; 
        display:block; 
        width:12px; 
        height:2px; 
        background:#f00;
        -webkit-transform: rotate(-90deg);
    }
}

.@{tag-prefix-cls}-selected {
    background: #FF9EAE;
    border: 1px solid #999;
}