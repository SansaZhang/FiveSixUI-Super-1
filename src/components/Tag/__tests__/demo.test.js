import React from 'react';
import Tag from '..';
import renderer from 'react-test-renderer';

// 简陋版GUI测试
// 在最开始需要撰写简单的应用场景，在开发、迭代时会提示对界面的更改，以确保更改是符合预期的


test('Tag demo', () => {
  const component = renderer.create(
    <Tag value={1}>子元素</Tag>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Tag demo', () => {
  const onTagClick = jest.fn();
  const key = 1000;
  const component = renderer.create(
    <Tag
      onClick={onTagClick}
      value={key}
      selected={true}
      closable={false}
    >
    label
    </Tag>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});