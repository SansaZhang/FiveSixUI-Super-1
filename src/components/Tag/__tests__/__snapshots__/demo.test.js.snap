exports[`test Tag demo 1`] = `
<div
  className="wl-tag"
  data-value={1}
  onClick={[Function]}>
  <span
    className="wl-tag-text">
    子元素
  </span>
  <span
    className="wl-tag-close"
    onClick={[Function]} />
</div>
`;

exports[`test Tag demo 2`] = `
<div
  className="wl-tag wl-tag-selected"
  data-value={1000}
  onClick={[Function]}>
  <span
    className="wl-tag-text">
    label
  </span>
  
</div>
`;
